### DISCUSSION TITLE
vcpkg_cmake_configure / vcpkg_cmake_install to add to PKG_CONFIG_PATH?

### DISCUSSION THREAD

**Discussion by RealLitb** [microsoft/vcpkg#32364] [Ideas]
**Describe the bug**

Currently the pkg config search path for CMake projects is found by using CMAKE_PREFIX_PATH (see documentation for FindPkgConfig.cmake). 

However, if a port uses CMake but builds vendored third-party projects of its own, these third party projects may use different build systems and simply assume, like Autoconfig, that the pkg-config search paths are in PKG_CONFIG_PATH. The qtwebengine port fails to build when the search paths are not listed in PKG_CONFIG_PATH. I don't know whether upstream is at fault for not ensuring that PKG_CONFIG_PATH has those entries, but upstream probably assumes that it's the builder's job to ensure that. 

VCPKG adds the correct search paths (depending on debug or non-debug config) for autoconfig ports. I believe it would be helpful if the cmake scripts could do that aswell.

**Comment by cenit** [comment]
same here.
Is it really by design that in order to make pkg-config "just work" inside portfile for cmake-based ports we need to add these lines?

```
vcpkg_find_acquire_program(PKGCONFIG)
set(ENV{PKG_CONFIG} "${PKGCONFIG}")
vcpkg_host_path_list(APPEND ENV{PKG_CONFIG_PATH} "${CURRENT_INSTALLED_DIR}/lib/pkgconfig")
```

shouldn't we just offer it to all ports, independently of cmake/make/meson toolchain adopted?
